src/__tests__/final/06.js

```diff
@@ -52,3 +52,37 @@ test('displays the users current location', async () => {
     `Longitude: ${fakePosition.coords.longitude}`,
   )
 })
+
+test('displays error message when geolocation is not supported', async () => {
+  const fakePosition = {
+    coords: {
+      latitude: 59,
+      longitude: 18,
+    },
+  }
+  const fakeError = new Error(
+    'Geolocation is not supported or permission denied',
+  )
+  const {promise, reject} = deferred()
+
+  window.navigator.geolocation.getCurrentPosition.mockImplementation(
+    (successCallback, errorCallback) => {
+      promise.then(
+        () => successCallback(fakePosition),
+        () => errorCallback(fakeError),
+      )
+    },
+  )
+
+  render(<Location />)
+
+  expect(screen.getByLabelText(/loading/i)).toBeInTheDocument()
+
+  await act(async () => {
+    reject()
+  })
```

Comment:
Let's make sure to wait for the promise to finish resolving before continuing:

```suggestion
  await act(async () => {
    reject()
    return promise
  })
```

Reply:
When I return the promise it doesn't work @kentcdodds 

<img width="630" alt="Screenshot 2020-10-30 at 19 56 27" src="https://user-images.githubusercontent.com/9104920/97746090-10b68a80-1aea-11eb-8064-77afe67d66d7.png">

Reply:
Oh, duh, we're rejecting the promise 🤦‍♂️

Haha, hmmm... Ok, I think we're fine. Thanks!